--- /usr/include/endian.h	2016-11-16 15:43:37.000000000 -0700
+++ build-sparse/include/endian.h	2017-03-15 12:43:28.732376898 -0600
@@ -59,38 +59,23 @@
 /* Conversion interfaces.  */
 # include <bits/byteswap.h>
 
-# if __BYTE_ORDER == __LITTLE_ENDIAN
-#  define htobe16(x) __bswap_16 (x)
-#  define htole16(x) (x)
-#  define be16toh(x) __bswap_16 (x)
-#  define le16toh(x) (x)
+// Simple vesion for sparse with annotations
+# include <util/compiler.h>
 
-#  define htobe32(x) __bswap_32 (x)
-#  define htole32(x) (x)
-#  define be32toh(x) __bswap_32 (x)
-#  define le32toh(x) (x)
+# define htobe16(x) ((__force __be16)__bswap_16 (x))
+# define htole16(x) ((__force __le16)__bswap_16 (x))
+# define be16toh(x) ((uint16_t)__bswap_16 ((__force uint16_t)(__be16)(x)))
+# define le16toh(x) ((uint16_t)__bswap_16 ((__force uint16_t)(__le16)(x)))
+
+# define htobe32(x) ((__force __be32)__bswap_32 (x))
+# define htole32(x) ((__force __le32)__bswap_32 (x))
+# define be32toh(x) ((uint32_t)__bswap_32 ((__force uint32_t)(__be32)(x)))
+# define le32toh(x) ((uint32_t)__bswap_32 ((__force uint32_t)(__le32)(x)))
+
+# define htobe64(x) ((__force __be64)__bswap_64 (x))
+# define htole64(x) ((__force __le64)__bswap_64 (x))
+# define be64toh(x) ((uint64_t)__bswap_64 ((__force uint64_t)(__be64)(x)))
+# define le64toh(x) ((uint64_t)__bswap_64 ((__force uint64_t)(__le64)(x)))
 
-#  define htobe64(x) __bswap_64 (x)
-#  define htole64(x) (x)
-#  define be64toh(x) __bswap_64 (x)
-#  define le64toh(x) (x)
-
-# else
-#  define htobe16(x) (x)
-#  define htole16(x) __bswap_16 (x)
-#  define be16toh(x) (x)
-#  define le16toh(x) __bswap_16 (x)
-
-#  define htobe32(x) (x)
-#  define htole32(x) __bswap_32 (x)
-#  define be32toh(x) (x)
-#  define le32toh(x) __bswap_32 (x)
-
-#  define htobe64(x) (x)
-#  define htole64(x) __bswap_64 (x)
-#  define be64toh(x) (x)
-#  define le64toh(x) __bswap_64 (x)
-# endif
 #endif
-
 #endif	/* endian.h */
